autogen/oai/together.py

```diff
@@ -0,0 +1,369 @@
+            params, "frequency_penalty", (int, float), True, None, (-2, 2), None
+        )
+        together_params["min_p"] = validate_parameter(params, "min_p", (int, float), True, None, (0, 1), None)
+        together_params["safety_model"] = validate_parameter(
+            params, "safety_model", str, True, None, None, None
+        )  # We won't enforce the available models as they are likely to change
+
+        # Check if they want to stream and use tools, which isn't currently supported (TODO)
+        if together_params["stream"] and "tools" in params:
+            warnings.warn(
+                "Streaming is not supported when using tools, streaming will be disabled.",
+                UserWarning,
+            )
+
+            together_params["stream"] = False
+
+        # To support avoidance of tool calling we allow the user to exclude tool calls on certain conditions
+        together_params["hide_tools"] = validate_parameter(
+            params, "hide_tools", str, False, "never", None, ["if_all_run", "if_any_run", "never"]
+        )
+
+        return together_params
+
+    def create(self, params: Dict) -> ChatCompletion:
+
+        messages = params.get("messages", [])
+
+        # Convert AutoGen messages to Together.AI messages
+        together_messages = oai_messages_to_together_messages(messages)
+
+        # Parse parameters to Together.AI API's parameters
+        together_params = self.parse_params(params)
+
+        # Handle tool use
+        tools = params.get("tools", None)
+        hide_tools = should_hide_tools(together_messages, tools, together_params["hide_tools"])
+
+        # We use chat model by default
+        client = Together(api_key=self.api_key)
+
+        # Token counts will be returned
+        prompt_tokens = 0
+        completion_tokens = 0
+        total_tokens = 0
+
+        max_retries = 5
+        for attempt in range(max_retries):
+            ans = None
+            try:
+                response = client.chat.completions.create(
+                    model=together_params["model"],
+                    max_tokens=together_params["max_tokens"],
+                    temperature=together_params["temperature"],
+                    top_p=together_params["top_p"],
+                    top_k=together_params["top_k"],
+                    repetition_penalty=together_params["repetition_penalty"],
+                    presence_penalty=together_params["presence_penalty"],
+                    frequency_penalty=together_params["frequency_penalty"],
+                    min_p=together_params["min_p"],
+                    stream=together_params["stream"],
+                    messages=together_messages,  # Main messages
+                    tool_choice=("auto" if not hide_tools else None),
+                    tools=tools if not hide_tools else None,  # Include any tools/functions
+                    n=1,  # API supports more than one response, however we will limit it to one
```

Comment:
Why? Can you check if OpenAIWrapper allows it?

Reply:
These are coming through the config and being applied to the call.